core/internal/cltest/simulated_backend.go

```diff
@@ -27,6 +27,28 @@ func NewSimulatedBackend(t *testing.T, alloc core.GenesisAlloc, gasLimit uint32)
 	})
 	return backend
 }
+func NewApplicationWithConfigV2OnSimulatedBlockchain(
+	t testing.TB,
+	cfg coreconfig.GeneralConfig,
+	backend *backends.SimulatedBackend,
+	flagsAndDeps ...interface{},
+) *TestApplication {
+	if bid := backend.Blockchain().Config().ChainID; bid.Cmp(testutils.SimulatedChainID) != 0 {
+		t.Fatalf("expected backend chain ID to be %s but it was %s", testutils.SimulatedChainID.String(), bid.String())
+	}
+	defID := cfg.DefaultChainID()
+	require.Zero(t, defID.Cmp(testutils.SimulatedChainID))
+	chainID := utils.NewBig(testutils.SimulatedChainID)
+	client := client.NewSimulatedBackendClient(t, backend, testutils.SimulatedChainID)
+	eventBroadcaster := pg.NewEventBroadcaster(cfg.DatabaseURL(), 0, 0, logger.TestLogger(t), uuid.NewV4())
+
+	flagsAndDeps = append(flagsAndDeps, client, eventBroadcaster, chainID)
+
+	//  app.Stop() will call client.Close on the simulated backend
+	app := NewApplicationWithConfig(t, cfg, flagsAndDeps...)
+
+	return app
+}
```

Comment:
Any way to avoid creating more of these helpers? Probably out of scope until we improve that API... :shrug:

Reply:
Unfortunately no, there was no other way to create an application without also automatically pulling in an EVM key, because we've always done it that way before.